pyproject.toml

```diff
@@ -1,6 +1,92 @@
 [build-system]
-requires = [
-    "setuptools>=42",
-    "wheel"
+requires = ["hatchling"]
+build-backend = "hatchling.build"
+
+[project]
+name = "undate"
+description = "library for working with uncertain, fuzzy, or partially unknown dates and date intervals"
+readme = "README.md"
+license = {text = "Apache-2"}
+requires-python = ">= 3.9"
+dynamic = ["version"]
+dependencies = [
+    "python-dateutil",
+    "lark"
 ]
-build-backend = "setuptools.build_meta"
+authors = [
+  {name = "Rebecca Sutton Koeser"},
+  {name = "Cole Crawford"},
+  {name = "Julia Damerow"},
+  {name = "Robert Casties"},
+  {name = "Malte Vogl"},
+#  {name = "DHTech", email="dhtech.community@gmail.com"}  ?
+]
+# currently no maintainers separate from authors
+
+keywords = ["dates", "dating", "uncertainty", "uncertain-dates", "unknown", "partially-known", "digital-humanities"]
+classifiers = [
+    "Development Status :: 2 - Pre-Alpha",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Intended Audience :: Developers",
+    "License :: OSI Approved :: Apache Software License",
+    "Operating System :: OS Independent",
+    "Topic :: Software Development :: Libraries :: Python Modules",
+    "Topic :: Utilities",
+    "Typing :: Typed"
+]
+
+[project.optional-dependencies]
+docs = [
+  "sphinx>=7.0.0",
+  "alabaster",
+  "myst-parser"
+]
+dev = [
+  "black>=22.10.0",
+  "pre-commit>=2.20.0",
+  "twine",
+  "wheel",
+  "build",
+  "mypy",
+  "treon",
+  "undate",
+  "undate[docs]",
+]
+test = [
+  "pytest>=7.2",
+  "pytest-ordering",
+  "pytest-cov",
+]
+all = [
+  "undate[dev]",
+  "undate[test]",
```

Comment:
CodeRabbit seems to be wrong about this - I like binding the two groups together into `all`